keps/sig-scheduling/5075-dra-consumable-capacity/README.md

```diff
@@ -0,0 +1,1533 @@
+to mark whether the device is shareable among multiple resource claims (or requests).
+The multi-allocatable device can be assigned to more than one request if it satisfies the selection criteria and constraints.
+The select condition `device.allowMultipleAllocations == true/false` is used to identify to select the device with a `allowMultipleAllocations` property or not.
+
+The enhancement also adds a `SharingPolicy` field to `DeviceCapacity`.
+This field specifies how the capacity can be shared between different requests.
+The sharing policy can either specify a range of valid values or a discrete set of them.
+Each policy has a default value, either explicitly or implicitly.
+
+Users can define specific per-device resource requests using the newly added `CapacityRequests` field,
+which is available in each supported device request type under `DeviceRequest`.
+Each `CapacityRequests` entry specifies the minimum required capacity for a device.
+If the capacity is consumable, the amount available for allocation is determined
+by subtracting the aggregated allocation results of current claims from the device's capacity as defined in the resource slice.
+The remaining amount will be used solely by the allocator and will not be reflected in the resource slice.
+The calculation of capacity requirements will round the requested capacity up to the nearest valid amount,
+based on the capacity's sharing policy.
+
+If users do not specify a capacity request for consumable capacity, the default consumed value will be applied.
+There is always such a default because capacities without a policy are not consumable.
+
+A device with `allowMultipleAllocations` property can only be allocated
+when its consumability has been verified and its attributes match the request's selectors and constraints.
+The newly added `consumedCapacities` field in the `DeviceRequestAllocationResult` will be set to the calculated capacity upon a successful allocation.
+This value may differ from the originally requested amount, as it is rounded up to the nearest valid value based on the device’s sharing policy.
+If no specific amount is requested, a default consumption value is applied.
+
+### API enhancement
+To enable this enhancement, the following API updates are proposed.
+
+#### ResourceSliceSpec's Device
+
+```go
+
+// Device represents one individual hardware instance that can be selected based
+// on its attributes. Besides the name, exactly one field must be set.
+type Device struct {
+...
+   // AllowMultipleAllocations marks whether the device is allowed to be allocated for multiple times.
+   //
+   // A device with allowMultipleAllocations="true" can be allocated more than once,
+   // and its capacity is shared, regardless of whether the CapacitySharingPolicy is defined or not.
+   //
+   // +optional
+   // +featureGate=DRAConsumableCapacity
+   AllowMultipleAllocations *bool
```

Comment:
xref prior thread for "enum vs bool": https://github.com/kubernetes/enhancements/pull/5104#discussion_r2094096057

I'm fine with a bool here because I cannot imagine other, future alternatives. A device is either shareable or it is not.

An enum makes a bit simpler to add new semantic if we clearly document "unknown value" as "ignore device", but we can get the same effect with a new field when being careful about how we roll out such a new alpha field (first ensure that all consumers know about it, then promote to beta). Because consumers are schedulers, this isn't too hard.